### ISSUE TITLE
Thank you for this!

### ISSUE THREAD

**Issue by asad-awadia** [typedpath/terraform2kotlin#1]
Thank you for starting work on this. What is the update rn? is it pretty much done or is it still under development?

My use cases would be pretty common - IAM roles, policies, dynamo, ec2 + ami and s3 etc

**Comment by typedpath** [comment]
no problem - i only started this a week ago so its probably at least a month off - i will put IAM roles, policies, dynamo db, ec2 , EC2/AMI, s3 at the top of the list along with my personal priority - kubernetes

**Comment by asad-awadia** [comment]
How are you implementing it? Maybe I can help

**Comment by typedpath** [comment]
I am planning a kotlin class for each terraform resource an data type: I committed classes for aws resources here yesterday : https://github.com/typedpath/terraform2kotlin/tree/master/terraform2kotlin/src/main/kotlin/com/typedpath/terraform2kotlin/aws/schema

**Comment by typedpath** [comment]
Hello, I have put some examples up covering your requests. For most of them I have also included client code to interact with the created infrastructure.
If you have any particular examples you are interested in let me know

**Comment by TanVD** [comment]
Just to let you know -- there is already a DSL for terraform inside Kotless (Kotlin Serverless Framework by JetBrains) [see](https://github.com/JetBrains/kotless/tree/master/engine/src/main/kotlin/io/kotless/terraform/provider/aws). Probably we can join our forces.

**Comment by chexov** [comment]
there is also this one which jetbrains uses now https://github.com/TanVD/terraform.kt

**Comment by TanVD** [comment]
Yep, we've moved part implemented in Kotless into a separate library :)
